api/impl/address.go

```diff
@@ -57,14 +57,19 @@ func (api *nodeAddrs) Lookup(ctx context.Context, addr address.Address) (peer.ID
 	return id, nil
 }
 
-func (api *nodeAddress) Import(ctx context.Context, d files.Directory) ([]address.Address, error) {
+func (api *nodeAddress) Import(ctx context.Context, f files.File) ([]address.Address, error) {
 	nd := api.api.node
 
-	kinfos, err := parseKeyInfos(d)
+	kinfos, err := parseKeyInfos(f)
 	if err != nil {
 		return nil, err
 	}
 
+	// error if we fail to parse keyinfo from the provided file.
+	if len(kinfos) == 0 {
+		return nil, fmt.Errorf("failed to parse keys from wallet file")
```

Comment:
Be more specific "no keys in wallet file %s"

Reply:
I will state that there are "no keys in the file", but I don't believe the interface for `go-ipfs-files` we are using has a `filename`/`string` methods, meaning I can't return the name of the file :(

Reply:
Aforementioned files interface: https://ipfs.io/ipfs/QmQmhotPUzVrMEWNK3x1R5jQ5ZHWyL7tVUrmRPjrBrvyCb/go-ipfs-files/file.go